docs/guides/mlflow.rst

```diff
@@ -1,18 +1,69 @@
-.. © Copyright 2023 Zapata Computing Inc.
+MLflow In Orquestra
+===================
 
-MLflow Utilities
-================
+Orquestra Platform comes with an `MLflow <https://mlflow.org/>`_ tracking server instance available for each workspace.
+To enable it, visit Orquestra Portal (your cluster's URL) > open a workspace > *Data* > click the *Connect* button.
+
+
+.. image:: images/portal-mlflow.png
+   :align: center
+
+
+Connecting From Python
+----------------------
+
+MLflow provides their own `Python client library <https://pypi.org/project/mlflow/>`_.
+Using a remote tracking server requires setting the tracking server URI and token.
+MLflow client expects these values to be present as environment variables.
+See also: `MLflow docs <https://mlflow.org/docs/latest/tracking.html#logging-to-a-tracking-server>`_.
+
+The Workflow SDK comes with utility functions that figure out the appropriate tracking URI and token for accessing the tracking server instance deployed on an Orquestra cluster.
+
+.. literalinclude:: ../examples/tests/test_mlflow_utilities.py
+    :start-after: def tracking_uri_token():
+    :end-before: </snippet>
+    :language: python
+    :dedent: 8
+
+When used on a local machine, the utility functions make use of the cluster URI and token stored in the config you created when :doc:`logging in<../tutorials/remote>`.
+
+When used in Orquestra Studio or within a task running on CE, the utility functions ignore the ``cluster_name=`` argument, and use a different mechanism for cluster-local URIs and auth.
+
+Passing in ``workspace_id`` to ``get_tracking_uri()`` is required regardless of the execution environment.
+
+
+Connecting From A Task
+----------------------
+
+The ``mlflow`` client reads auth URI and token from environment variables.
```

Comment:
Not urgent, but do you know which grammar applies here, making the definite article required?

Reply:
The distinction here is between talking about a member of a category, and a category itself. We need the article to clarify grammatically that we're talking about a specific auth URI, rather than the concept of auth URIs. 

[omitted for brevity: a long digression about Plato and the world of forms]

There's a significant difference between the way a name operates, and the way a description does. The sentence would be fine if it was "a and b" or "Carlos and Simon" because these are labels that uniquely identify entities of a known type, variables in the first case and people in the second. They convey no information about what classes of thing a, b, Carlos and Simon belong to, the reader's knowledge and the context take care of that so all the name does is uniquely identify the subject - the reader already knows how a person is quantised and that Carlos is a person so we don't need the article. The reader's brain effectively reads it as "Carlos the person".

Here, however, "auth URI" has to tell you _what_ it is at the same time as telling you, effectively, _who_ it is. As such, we need the article to tell the reader that we're talking about a member of the category "auth URI" rather than the category itself. In the absence of the article, the default reading is that we're talking about an indeterminate quantity of _stuff_ that belongs to the category. As an example, consider the difference between "There is a deer on the tracks" and "There is deer on the tracks". This is why it's fine to omit or include the article for "environment variables" - for the purpose of this sentence, the abstract class of "environment variables" and the concrete members "the environment variables" are both valid.

The choice of article is less concrete, we use the indefinite article here and still be grammatically accurate. However the definite article is the appropriate choice in context - the client is reading a specific token whose identity is known in advance.